src/irc.py

```diff
@@ -139,42 +175,82 @@ def _write(self, path: pathlib.Path, source: str) -> None:
         path.write_text(source, encoding="utf-8")
         logger.info("Pushed file %s", path)
 
-    def _generate_PEM_file_local(self) -> str:
+    def _generate_PEM_file_local(self) -> None:
         """Generate the PEM file content.
 
-        Returns:
-            A string
+        Raises:
+            
         """
-        pass
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem ]] || "
+            + "openssl genpkey -out {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem "
+            + "-outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048",
+        ]
+        logger.info("Creating PEM file for IRC bridge.")
+        exec_process = subprocess.run(
+           pem_create_command, shell=True, check=True, capture_output=True)
+        )
+        logger.info("PEM create output: %s.", exec_process.stdout)
 
-    def _generate_conf_local(self) -> str:
-        """Generate the content of the irc configuration file.
+    def _generate_app_registration_local(self) -> str:
+        """Generate the content of the app registration file.
 
         Returns:
             A string
         """
-        pass
+        app_reg_create_command = [
+                "/bin/bash",
+                "-c",
+                f"[[ -f {IRC_BRIDGE_REGISTRATION_PATH} ]] || "
+                f"/bin/node /app/app.js -r -f {IRC_BRIDGE_REGISTRATION_PATH} "
+                f"-u http://localhost:{IRC_BRIDGE_HEALTH_PORT} "
```

Comment:
If I remember well this will declare the IRC Bridge IP, localhost worked when running the bridge and synapse on the same machine, but now that won't work, so it should declare the unit IP address here (beware of scaling where there could be multiple IP)